File: backend/app/api/routes/stats.py
from fastapi import APIRouter, HTTPException

from app.schemas.stats import AnomalyRequest, BurdenRequest, ClusterRequest, CorrelateRequest, TrendRequest
from app.services.stats_engine import (
    cluster_countries,
    compute_burden_index,
    compute_correlation,
    compute_trend,
    detect_anomalies,
)

router = APIRouter(prefix="/stats", tags=["stats"])


def _handle(fn, *args, **kwargs):
    try:
        return fn(*args, **kwargs)
    except ValueError as exc:
        raise HTTPException(status_code=422, detail=str(exc))
    except Exception as exc:
        raise HTTPException(status_code=500, detail=f"Computation error: {exc}")


@router.post("/trend")
def trend(body: TrendRequest):
    return _handle(compute_trend, body.records, body.project_years)


@router.post("/correlate")
def correlate(body: CorrelateRequest):
    return _handle(compute_correlation, body.x_records, body.y_records)


@router.post("/burden")
def burden(body: BurdenRequest):
    return _handle(compute_burden_index, body.records, body.weights, body.invert)


@router.post("/anomalies")
def anomalies(body: AnomalyRequest):
    return _handle(detect_anomalies, body.records, body.window)


@router.post("/cluster")
def cluster(body: ClusterRequest):
    return _handle(cluster_countries, body.records, body.n_clusters)

File: frontend/src/api/stats.js
import api from './auth'

export async function getTrend(records, projectYears = 5) {
  const { data } = await api.post('/stats/trend', { records, project_years: projectYears })
  return data
}

export async function getAnomalies(records, window = 5) {
  const { data } = await api.post('/stats/anomalies', { records, window })
  return data
}

export async function getCorrelation(xRecords, yRecords) {
  const { data } = await api.post('/stats/correlate', { x_records: xRecords, y_records: yRecords })
  return data
}

export async function getBurden(records, weights = null, invert = null) {
  const { data } = await api.post('/stats/burden', { records, weights, invert })
  return data
}

export async function getCluster(records, nClusters = 4) {
  const { data } = await api.post('/stats/cluster', { records, n_clusters: nClusters })
  return data
}
